pandas/_libs/hashtable_class_helper.pxi.in

```diff
@@ -792,9 +789,9 @@ cdef class StringHashTable(HashTable):
                 labels[i] = na_sentinel
             else:
                 # if ignore_na is False, we also stringify NaN/None/etc.
-                v = get_c_string(<str>val)
-                if v == NULL:
-                    PyErr_Clear()
+                try:
+                    v = get_c_string(<str>val)
+                except UnicodeEncodeError:
```

Comment:
do we have anything in the test suite that hits this now? (if we do can you add this issue number there)

Reply:
This issue was triggered by the following test: pandas/tests/frame/test_api.py::test_column_name_contains_unicode_surrogate
Which was added because of a fix for #25509